lib/cypress/population_clone_job.rb

```diff
@@ -72,13 +72,18 @@ def randomize_ids(patients, prng)
 
     # if provider argument is nil, this function will assign a new provider based on the @option['providers'] and @option['generate_provider'] options
     def clone_and_save_patient(patient, prng, provider = nil, allow_dups = false)
-      cloned_patient = patient.clone
+      # This operates on the assumption that we are always cloning a patient for a product test.
+      # If we need to clone a patient for any other reason then we will need to paramaterize
+      # the type coming into this class.
+      cloned_patient = ProductTestPatient.new(patient.attributes.except(:_id, :_type))
+      cloned_patient.attributes = {
+        original_medical_record_number: patient.medical_record_number,
+        original_patient_id: patient.id,
+        correlation_id: options['test_id']
+      }
       # TODO: R2P: use patient model
```

Comment:
Can this `TODO` be removed now?

Reply:
I believe so, I've gone ahead and removed it.